Sources/NIOHTTP1/HTTPObjectAggregator.swift

```diff
@@ -0,0 +1,422 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// The parts of a complete HTTP response from the view of the client.
+///
+/// Afull HTTP request is made up of a response header encoded by `.head`
+/// and an optional `.body`.
+public struct HTTPServerRequestFull {
+    public var head: HTTPRequestHead
+    public var body: ByteBuffer?
+
+    public init(head: HTTPRequestHead, body: ByteBuffer?) {
+        self.head = head
+        self.body = body
+    }
+}
+
+extension HTTPServerRequestFull: Equatable {}
+
+/// The parts of a complete HTTP response from the view of the client.
+///
+/// Afull HTTP response is made up of a response header encoded by `.head`
+/// and an optional `.body`.
+public struct HTTPClientResponseFull {
+    public var head: HTTPResponseHead
+    public var body: ByteBuffer?
+
+    public init(head: HTTPResponseHead, body: ByteBuffer?) {
+        self.head = head
+        self.body = body
+    }
+}
+
+extension HTTPClientResponseFull: Equatable {}
+
+public enum HTTPObjectAggregatorError: Error {
```

Comment:
So we probably need to replace these errors with a custom struct error. This can carry an error code, but that error code cannot be an `enum`.